code/code/sys/indexdata.cc

```diff
@@ -210,181 +149,118 @@ void generate_obj_index()
   /********************/
 
   TDatabase db(DB_SNEEZY);
-  db.query("select vnum, name, short_desc, long_desc, max_exist, spec_proc, weight, max_struct, wear_flag, type, price, action_desc from obj order by vnum");
-
-  while(db.fetchRow()){
-    tmpi = new objIndexData();
-    if (!tmpi) {
-      perror("indexData");
-      exit(0);
-    }
-    
-    tmpi->virt=convertTo<int>(db["vnum"]);
-    tmpi->name=mud_str_dup(db["name"]);
-    tmpi->short_desc=mud_str_dup(db["short_desc"]);
-    tmpi->long_desc=mud_str_dup(db["long_desc"]);
-    tmpi->max_exist=convertTo<int>(db["max_exist"]);
-
-    // use 327 so we don't go over 32765 in calculation
-    if (tmpi->max_exist < 327) {
-      tmpi->max_exist *= (short) (stats.max_exist * 100);
-      tmpi->max_exist /= 100;
-    }
-    if (tmpi->max_exist)
-      tmpi->max_exist = max(tmpi->max_exist, (short int) 1);
-    
-
-    tmpi->spec=convertTo<int>(db["spec_proc"]);
-    tmpi->weight=convertTo<float>(db["weight"]);
-    tmpi->max_struct=convertTo<int>(db["max_struct"]);
-    tmpi->where_worn=convertTo<int>(db["wear_flag"]);
-    tmpi->itemtype=convertTo<int>(db["type"]);
-    tmpi->value=convertTo<int>(db["price"]);
-    if(!db["action_desc"].empty())
-      tmpi->description=mud_str_dup(db["action_desc"]);
-    else tmpi->description=NULL;
-
-    while(!extra_db["vnum"].empty() && convertTo<int>(extra_db["vnum"]) < tmpi->virt){
+  db.query(
+      "select vnum, name, short_desc, long_desc, max_exist, spec_proc, weight, max_struct, wear_flag, type, price, action_desc from obj order by vnum");
+
+  while (db.fetchRow()) {
+    int virt = convertTo<int>(db["vnum"]);
+    sstring name = db["name"];
+    sstring short_desc = db["short_desc"];
+    sstring long_desc = db["long_desc"];
+
+    int max_exist = convertTo<int>(db["max_exist"]);
+    max_exist = static_cast<int>(max_exist * stats.max_exist);
+    max_exist = min(max(max_exist, 1), 9999);
+
+    int spec = convertTo<int>(db["spec_proc"]);
+    float weight = convertTo<float>(db["weight"]);
+    byte max_struct = convertTo<int>(db["max_struct"]);
+    unsigned int where_worn = convertTo<int>(db["wear_flag"]);
+    ubyte itemtype = convertTo<int>(db["type"]);
+    int value = convertTo<int>(db["price"]);
+    sstring description = db["action_desc"];
+
+    extraDescription *ex_description = nullptr;
+    while (!extra_db["vnum"].empty() && convertTo<int>(extra_db["vnum"]) <= virt) {
+      if (convertTo<int>(extra_db["vnum"]) == virt) {
+        auto new_descr = new extraDescription();
+        new_descr->keyword = extra_db["name"];
+        new_descr->description = extra_db["description"];
+        new_descr->next = ex_description;
+        ex_description = new_descr;
```

Comment:
Eyebleed. Not in the scope of your PR, perhaps, but I'd enjoy making it a vector of strings.

Reply:
I worked on converting ex_description to a std::list for a bit, but it ended up just feeling like way too big of a project for way too little gain, as it's used all over. Decided just to leave it as-is, and maybe tackle it in a future PR as its own thing.